packages/block-library/src/navigation-link/edit.js

```diff
@@ -278,6 +279,7 @@ function NavigationLinkEdit( {
 											return escape( normalizedURL );
 										} )(),
 										opensInNewTab: newOpensInNewTab,
+										type,
```

Comment:
These suggestion types are proposed to be removed by the issue at #20051 . Is it strictly necessary for them to exist in this implementation, in mind of the problems and alternatives considered in #20051 ?

Reply:
It depends. I don't see a problem with having type as `LinkControl` does create many types of items which depend on the, well, type of suggestion picked by the user. In this context as you can see it is extremely useful to have it as it becomes easy to render dynamic links if the URL is not manually entered.

If we remove the type we'd check for id then. But it does seem like a good descriptor to have since we actually do have many types and we treat them differently. I am not a fan of inferring categories from data values when I can store a flat thing that tells me what it is.

Reply:
The problem for me is that these types are neither clearly documented nor based on some shared understanding of categorization of links. While they're not _wrong_, they feel arbitrary. My fear is that this will cause more confusion than it will help, both in that a consumer will not clearly understand which options are available to them (and how they relate to other "types" within WordPress like a post type), and that a future maintainer will not have confidence to make a decision in adding to or revising these types. Furthermore, if it is the case that these types can be derived based on other values (like `id`), then they can be considered redundant, which leads to additional problems of keeping values in sync when one changes.

I'd not mind if there were to be some utility functions which abstract this in a way which is more human-readable, but generally I think the further we can get with the fewest distinct attribute values, the better.

Reply:
I removed the "type" checking and also fixed `LinkControl` to not set an id for manual entry URLs. :) \o/